apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/RestoreBucketTest.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Storage.V1.IntegrationTests
+{
+    [Collection(nameof(StorageFixture))]
+    public class RestoreBucketTest
+    {
+        private readonly StorageFixture _fixture;
+
+        public RestoreBucketTest(StorageFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Fact]
+        public async Task RestoreSoftDeletedBucket()
+        {
+
+            // We get bucket object using get bucket async method for soft delete bucket before deleting it. 
+            var bucket = await _fixture.Client.GetBucketAsync(_fixture.SoftDeleteBucketThree, new GetBucketOptions { SoftDeleted = false });
+            // We delete soft delete bucket.
+            try
+            {
+                await _fixture.Client.DeleteBucketAsync(_fixture.SoftDeleteBucketThree);
+            }
+            catch (Exception)
+            {
+                // If bucket is not empty, we delete on a best effort basis.
+                foreach (var storageObject in _fixture.Client.ListObjects(_fixture.SoftDeleteBucketThree, ""))
+                {
+                    await _fixture.Client.DeleteObjectAsync(_fixture.SoftDeleteBucketThree, storageObject.Name);
+                }
+                await _fixture.Client.DeleteBucketAsync(_fixture.SoftDeleteBucketThree);
+            }
+
+            // And now we can restore bucket using generation id.
```

Comment:
In general avoid living comments that simply describe the code that follows, unless it's very complicated or particular code.